contracts/protocol/libraries/logic/ReserveLogic.sol

```diff
@@ -296,19 +298,18 @@ library ReserveLogic {
         reserveCache.currLiquidityIndex
       );
       reserve.liquidityIndex = reserveCache.nextLiquidityIndex.toUint128();
+    }
 
-      //as the liquidity rate might come only from stable rate loans, we need to ensure
-      //that there is actual variable debt before accumulating
-      if (reserveCache.currScaledVariableDebt != 0) {
-        uint256 cumulatedVariableBorrowInterest = MathUtils.calculateCompoundedInterest(
-          reserveCache.currVariableBorrowRate,
-          reserveCache.reserveLastUpdateTimestamp
-        );
-        reserveCache.nextVariableBorrowIndex = cumulatedVariableBorrowInterest.rayMul(
-          reserveCache.currVariableBorrowIndex
-        );
-        reserve.variableBorrowIndex = reserveCache.nextVariableBorrowIndex.toUint128();
-      }
+    // Variable borrow side only gets updated if there is any accrual of variable debt
```

Comment:
This comment seems untrue to me. You are using `currScaledVariableDebt` which is the total variable debt rather than the accrual

Reply:
Agree, more precise now here https://github.com/aave/aave-v3-core/pull/724/commits/efdaf634a7809e254ae27c2543f3c64ac875d83e